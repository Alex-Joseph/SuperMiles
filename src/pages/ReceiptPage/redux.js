import { logoutUser } from '../LoginPage/actions';


export const mapStateToProps = state => ({
  loggedIn: state.toJS().auth.loggedIn,
  balance: state.toJS().wallet.balance,
  transaction: state.toJS().wallet.transaction,
});

export const mapDispatchToProps = {
  logoutUser,
};
