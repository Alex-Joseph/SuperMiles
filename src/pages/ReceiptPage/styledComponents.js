import styled from 'styled-components';


const StyledMarketPlaceWrapper = styled.div`
  display: grid;
  place-content: center;
  padding: 24px 12px;

  h1 {
    text-align: center;
  }
`;

export { StyledMarketPlaceWrapper };
