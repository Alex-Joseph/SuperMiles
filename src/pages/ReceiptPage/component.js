import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { Header, Image, Button } from 'semantic-ui-react';
import { StyledMarketPlaceWrapper } from './styledComponents';
import Navbar from '../../components/Navbar';
import AirBudsBlue from '../../images/air_buds_blue.jpg';
import AirBudsBlack from '../../images/air_buds_black.jpg';


const imageMap = {
  'black': AirBudsBlack,
  'blue': AirBudsBlue,
};

const ReceiptPage = ({ balance, logoutUser, transaction, loggedIn, history }) => (
  <Fragment>
    <Navbar title="SuperMiles" />
    <StyledMarketPlaceWrapper>
      <Header as='h1'>New Balance: {balance}</Header>
      <Image src={imageMap[transaction.desc]} size='medium' rounded />
      <Button onClick={() => logoutUser(history)}>Logout</Button>
    </StyledMarketPlaceWrapper>
  </Fragment>
);

ReceiptPage.propTypes = {
  balance: PropTypes.number.isRequired,
  logoutUser: PropTypes.func.isRequired,
  history: PropTypes.shape({}).isRequired,
  transaction: PropTypes.shape({}).isRequired,
};

ReceiptPage.defaultProps = {
  transaction: {},
};

export default ReceiptPage
