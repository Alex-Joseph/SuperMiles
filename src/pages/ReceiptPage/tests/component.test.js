import React from 'react';
import { shallow } from 'enzyme';
import ReceiptPage from '../component';


describe('ReceiptPage', () => {
  const balance = 10000;
  const logoutUser = jest.fn();
  const push = jest.fn();
  const transaction = {desc: 'black'};

  it('renders correct snapshot', () => {
    const wrapper = shallow(
      <ReceiptPage
        balance={balance}
        logoutUser={logoutUser}
        transaction={transaction}
        history={{push}}
      />,
    );
    const expectedImageSrc = 'air_buds_black.jpg';

    expect(wrapper.find('Header').props().children[1]).toEqual(balance);
    expect(wrapper.find('Image').props().src).toEqual(expectedImageSrc);
    expect(wrapper).toMatchSnapshot();
  });

  it('calles logoutUser when the Logout Button is clicked', () => {
    const wrapper = shallow(
      <ReceiptPage
        balance={balance}
        logoutUser={logoutUser}
        transaction={transaction}
        history={{push}}
      />,
    );

    wrapper.find('Button').simulate('click');

    expect(logoutUser).toHaveBeenCalledWith({push});
  });
});
