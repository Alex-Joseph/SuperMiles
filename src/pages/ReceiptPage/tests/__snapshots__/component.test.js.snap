// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReceiptPage renders correct snapshot 1`] = `
<Fragment>
  <Navbar
    title="SuperMiles"
  />
  <styled.div>
    <Header
      as="h1"
    >
      New Balance: 
      10000
    </Header>
    <Image
      as="img"
      rounded={true}
      size="medium"
      src="air_buds_black.jpg"
      ui={true}
    />
    <Button
      as="button"
      onClick={[Function]}
    >
      Logout
    </Button>
  </styled.div>
</Fragment>
`;
